File: app/src/main/java/com/akhbulatov/githubsample/data/global/DataManager.java
package com.akhbulatov.githubsample.data.global;

import android.support.annotation.NonNull;

import com.akhbulatov.githubsample.models.User;
import com.akhbulatov.githubsample.models.UserDetails;

import java.util.List;

import retrofit2.Call;

public interface DataManager {
    Call<List<User>> getUsers();

    Call<UserDetails> getUserDetails(@NonNull String login);
}

File: app/src/main/java/com/akhbulatov/githubsample/data/global/DataManager.kt
package com.akhbulatov.githubsample.data.global

import com.akhbulatov.githubsample.models.User
import com.akhbulatov.githubsample.models.UserDetails

import retrofit2.Call

interface DataManager {
    fun getUsers(): Call<List<User>>
    fun getUserDetails(login: String): Call<UserDetails>

    fun getFavorites(): List<UserDetails>
    fun addUserToFavorites(userDetails: UserDetails)
}